src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java

```diff
@@ -93,4 +96,23 @@ public void setSwipingStatus(long activityId, String token, SwipeStatus swipeSta
 
         return generatedActivities;
     }
+
+    public List<Activity> getAllActivitiesOfUser(User user){
+        return activityRepository.findByUserSwipeStatusList_User(user);
+    }
+
+    public List<Activity> getAllActivitiesWithMatchedUsers(User user){
+        ActivityService activityService = new ActivityService(activityRepository,activityPresetRepository,userService);
+        List<Activity> allActivitiesWithMatchedUsers;
+
+        List<Activity> allActivitiesOfUser = activityService.getAllActivitiesOfUser(user);
```

Comment:
Why do you create a ActivityService inside the ActivityService class?

How about:
```suggestion
        List<Activity> allActivitiesWithMatchedUsers;
        List<Activity> allActivitiesOfUser = getAllActivitiesOfUser(user);
```